docs/source/dialogs.md

```diff
@@ -65,3 +69,23 @@ def process(self):
         ips = [entry[4][0] for entry in resolved if not '%' in entry[4][0]]
         self.produce(ResolvedHostname(name=hostname, ips=ips))
 ```
+
+### Explaining the dialogs processing mechanism during the upgrade
```

Comment:
The title is suggesting something different from the text below. Basically, regarding the text below, it should be something like `Explaining the upgrade workflow`, if you know what I mean. In case it should be only about the dialogs, change text below, but I prefer to change the title.